### PR TITLE
storage: add test cases with zero-sized object IDs

### PR DESCRIPTION
**Pull request by jforissier**
The TEE Internal Core API specification (v1.3.1) explicitly allows the
use of an empty object ID in TEE_RenamePersistentObject(). The text is:

 newObjectID, newObjectIDLen: A buffer containing the new object
   identifier. The identifier contains arbitrary bytes, including the
   zero byte. The identifier length SHALL be less than or equal to
   TEE_OBJECT_ID_MAX_LEN and can be zero.

(note the mention: "and can be zero").

Therefore, add a couple of test cases in regression_6008. Both
(non-NULL, 0) and (NULL, 0) are tested for the parameters
(newObjectID, newObjectIDLen).

Since it follows from the above that zero-sized object IDs are indeed
valid, similar test cases are added to regression_6001
(TEE_CreatePersistentObject()) and regression_6002
(TEE_OpenPersistentObject()).

Signed-off-by: Jerome Forissier <jerome@forissier.org>

### CODE REVIEW

**Comment by jforissier** [comment]
Depends on https://github.com/OP-TEE/optee_os/pull/5188.

**Review by etienne-lms — COMMENTED** [review]
Should there be a failling test where oid is NULL and oid-length is non-zero? the test should gently refuse the object creation or renaming.

File: host/xtest/regression_6000.c [hunk 1]
```diff
@@ -999,6 +1051,52 @@ static void xtest_tee_test_6008_single(ADBG_Case_t *c, uint32_t storage_id)
...
+			TEE_DATA_FLAG_ACCESS_READ |
+			TEE_DATA_FLAG_ACCESS_WRITE_META, &obj, storage_id)))
+		goto exit;
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c, fs_read(&sess, obj, out, 10, &count)))
+		goto exit;
+
+	(void)ADBG_EXPECT_BUFFER(c, data_00, 10, out, count);
+
+	/* Object ID = (non-NULL, 0) */
```
[the remark this answers is not in the document]
Comment (jforissier) [reply host/xtest/regression_6000.c:1075]: Right, will fix

**Comment by jforissier** [comment]
> Should there be a failling test where oid is NULL and oid-length is non-zero? the test should gently refuse the object creation or renaming.

Yes, maybe. In fact when adding such a test I noticed that the NULL pointer is not actually passed down to the TEE Internal API: there is a `TEE_Malloc()` in the storage TA so I need to fix this.

**Comment by jforissier** [comment]
Update:
- Added commit "ta/storage: allow to pass a NULL objectId to open, create and rename" (I will move it before "storage: add test cases with zero-sized object IDs" when rebasing)
- Fixed comment

I didn't add any test with object ID (NULL, non-zero) because it is a bit difficult to do. The reason is the buffer is passed from the CA to the TA by `TEE_InvokeCommand()` in a `tmpref` parameter, and this invalid combination is rejected with `TEE_ERROR_INVALID_PARAMETER` so it doesn't even reach the TA. Some more code would be needed (such as a special command for the storage TA perhaps).

**Comment by etienne-lms** [comment]
Right, it maybe not worth the pain (maintainance). There no such test neither on the data part.

**Comment by jenswi-linaro** [comment]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`
